controllers/eventingauth_controller.go

```diff
@@ -165,6 +178,39 @@ func (r *eventingAuthReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}, nil
 }
 
+func (r *eventingAuthReconciler) getIasClient() (ias.Client, error) {
+	namespace, name := getIasSecretNamespaceAndNameConfigs()
+	newIasCredentials, err := ias.ReadCredentials(*namespace, *name, r.Client)
+	if err != nil {
+		return nil, err
+	}
+	// return from cache unless credentials are changed
+	if reflect.DeepEqual(newIasCredentials, r.iasCredentials) {
+		return r.iasClient, nil
+	}
+	// update ias client if credentials are changed
+	r.iasCredentials = *newIasCredentials
+	iasClient, err := ias.NewIasClient(newIasCredentials.URL, newIasCredentials.Username, newIasCredentials.Password)
+	if err != nil {
+		return nil, fmt.Errorf("failed to createa a new IAS client: %v", err)
+	}
+	// caching ias client
+	r.iasClient = iasClient
+	return iasClient, nil
+}
+
+func getIasSecretNamespaceAndNameConfigs() (*string, *string) {
```

Comment:
Let's document local development very well, with proper commands so that a new developer gets started fast. 
The we have a consistent way of getting credentials only from secret. I'll delete env vars then.